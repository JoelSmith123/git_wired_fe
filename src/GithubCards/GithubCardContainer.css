
.GithubCardsContainer {
  width:            35%;
  display:          flex;
  flex-direction:   column;
  flex-wrap:        nowrap;
  vertical-align:   top;
  align-content:    space-around;
  align-items:      stretch;
  background-color: #eaf2f2
}

.GithubTitle {
  font-size:     3vw;
  text-align:    center;
}
