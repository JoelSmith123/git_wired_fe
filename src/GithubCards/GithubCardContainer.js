
import React, { Component } from 'react';

import './GithubCardContainer.css'

import GithubCard from './GithubCard.js'



export default class GithubCardContainer extends Component {
  constructor() {
    super();
    this.state = {
      showDropdown: false,
      statusName: ''
    }
  }

  // TO DO - TEST ME
  renderGithubCards = () => {
    let cards = this.props.cards
    return cards.map( (card, index) => <GithubCard card={card} key={index} /> )
  }

  toggleDropdown = (event) => {
    event.preventDefault()

    this.setState({ showDropdown: !this.state.showDropdown })
  }

  cardsStatusSelect = (event, statusName) => {
    event.preventDefault()

    this.props.filterCardsAndTemplateElementsByStatus(statusName)
    this.setState({ showDropdown: false, statusName })
  }

  // TO DO - TEST ME
  render() {
    return (
      <div className='github-cards-container'>
        <div className='github-cards-container-header'>
          <div className='github-title'>Github Issues</div>
          <button onClick={this.toggleDropdown}>status</button>
          {
            this.state.showDropdown ? (
              <div className='github-cards-container-status-dropdown'>
                <button className='github-cards-container-status-dropdown-btn'
                        onClick={(event) => this.cardsStatusSelect(event, event.target.name)}
                >---Show All---</button>
                <button className='github-cards-container-status-dropdown-btn'
                        name='open'
                        onClick={(event) => this.cardsStatusSelect(event, event.target.name)}
                >Open</button>
                <button className='github-cards-container-status-dropdown-btn'
                        name='closed'
                        onClick={(event) => this.cardsStatusSelect(event, event.target.name)}
                >Closed</button>
                <button className='github-cards-container-status-dropdown-btn'
                        name='in-progress'
                        onClick={(event) => this.cardsStatusSelect(event, event.target.name)}
                >In-Progress</button>
                <button className='github-cards-container-status-dropdown-btn'
                        name='to-do'
                        onClick={(event) => this.cardsStatusSelect(event, event.target.name)}                        
                >To Do</button>
              </div>
            ) : (
              null
            )
          }
        </div>
        <div className='github-cards'>
          { this.renderGithubCards() }
        </div>
      </div>
    )
  }


}
