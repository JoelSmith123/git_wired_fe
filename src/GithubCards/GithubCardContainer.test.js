import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import GithubCardContainer from './GithubCardContainer.js'

describe('GithubCardContainer', () => {
  let data1
  let data2
  let cards
  beforeEach(() => {
    data1 = {
       'cardTitle':       'This is the title',
       'cardDescription': 'This is lots of long text from the body of my issue',
       'cardNumber':       23,
       'cardURL':         'https://github.com/JoelSmith123/git_wired_fe/issues/23',
       'cardStatus':      'Open',
     }

    data2 = {
       'cardTitle':       'This is title2',
       'cardDescription': 'Description 2',
       'cardNumber':       16,
       'cardURL':         'https://github.com/JoelSmith123/git_wired_fe/issues/16',
       'cardStatus':      'Open',
     }

     cards = [ data1, data2]
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GithubCardContainer cards={ cards } />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<GithubCardContainer cards={ cards }/>)
    expect(wrapper).toMatchSnapshot()
  })   

  it('should render github cards', () => {
    const wrapper = shallow(<GithubCardContainer cards={ cards }/>)

    expect(wrapper.instance().renderGithubCards()).toHaveLength(2)
  })

  it('should toggle dropdown state', () => {
    const wrapper = shallow(<GithubCardContainer cards={ cards }/>)
    const mockEvent = { target: {}, preventDefault: jest.fn() }

    wrapper.setState({ showDropdown: false })
    wrapper.instance().toggleDropdown(mockEvent)

    expect(wrapper.state('showDropdown')).toEqual(true)
  })

  it('should set state for statusName and change dropdown to false', () => {
    const wrapper = shallow(<GithubCardContainer cards={ cards } filterCardsAndTemplateElementsByStatus={jest.fn()}/>)
    const mockEvent = { target: {}, preventDefault: jest.fn() }
    const statusName = 'status'

    wrapper.setState({ showDropdown: true, statusName: ''})
    wrapper.instance().cardsStatusSelect(mockEvent, statusName)

    expect(wrapper.state('showDropdown')).toEqual(false)
    expect(wrapper.state('statusName')).toEqual(statusName)
  })

  it('should render dropdown on state change', () => {
    const wrapper = mount(<GithubCardContainer cards={ cards } filterCardsAndTemplateElementsByStatus={jest.fn()}/>)

    wrapper.setState({ showDropdown: true })

    expect(wrapper.find('.github-cards-container-status-dropdown')).toHaveLength(1)
  })

  it('should not render dropdown on state change', () => {
    const wrapper = mount(<GithubCardContainer cards={ cards } filterCardsAndTemplateElementsByStatus={jest.fn()}/>)

    wrapper.setState({ showDropdown: false })

    expect(wrapper.find('.github-cards-container-status-dropdown')).toHaveLength(0)    
  })

  it('should render github cards inside element', () => {
    const wrapper = mount(<GithubCardContainer cards={ cards } filterCardsAndTemplateElementsByStatus={jest.fn()}/>)

    expect(wrapper.find('.github-cards').children()).toHaveLength(2)
  })

});
