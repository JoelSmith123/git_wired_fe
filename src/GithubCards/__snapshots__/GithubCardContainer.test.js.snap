// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubCardContainer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GithubCardContainer
    cards={
      Array [
        Object {
          "cardDescription": "This is lots of long text from the body of my issue",
          "cardNumber": 23,
          "cardStatus": "Open",
          "cardTitle": "This is the title",
          "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
        },
        Object {
          "cardDescription": "Description 2",
          "cardNumber": 16,
          "cardStatus": "Open",
          "cardTitle": "This is title2",
          "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/16",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="github-cards-container-header"
        >
          <div
            className="github-title"
          >
            Github Issues
          </div>
          <button
            onClick={[Function]}
          >
            status
          </button>
        </div>,
        <div
          className="github-cards"
        >
          <GithubCard
            card={
              Object {
                "cardDescription": "This is lots of long text from the body of my issue",
                "cardNumber": 23,
                "cardStatus": "Open",
                "cardTitle": "This is the title",
                "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
              }
            }
          />
          <GithubCard
            card={
              Object {
                "cardDescription": "Description 2",
                "cardNumber": 16,
                "cardStatus": "Open",
                "cardTitle": "This is title2",
                "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/16",
              }
            }
          />
        </div>,
      ],
      "className": "github-cards-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="github-title"
            >
              Github Issues
            </div>,
            <button
              onClick={[Function]}
            >
              status
            </button>,
            null,
          ],
          "className": "github-cards-container-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Github Issues",
              "className": "github-title",
            },
            "ref": null,
            "rendered": "Github Issues",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "status",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "status",
            "type": "button",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <GithubCard
              card={
                Object {
                  "cardDescription": "This is lots of long text from the body of my issue",
                  "cardNumber": 23,
                  "cardStatus": "Open",
                  "cardTitle": "This is the title",
                  "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
                }
              }
            />,
            <GithubCard
              card={
                Object {
                  "cardDescription": "Description 2",
                  "cardNumber": 16,
                  "cardStatus": "Open",
                  "cardTitle": "This is title2",
                  "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/16",
                }
              }
            />,
          ],
          "className": "github-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "card": Object {
                "cardDescription": "This is lots of long text from the body of my issue",
                "cardNumber": 23,
                "cardStatus": "Open",
                "cardTitle": "This is the title",
                "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "card": Object {
                "cardDescription": "Description 2",
                "cardNumber": 16,
                "cardStatus": "Open",
                "cardTitle": "This is title2",
                "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/16",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="github-cards-container-header"
          >
            <div
              className="github-title"
            >
              Github Issues
            </div>
            <button
              onClick={[Function]}
            >
              status
            </button>
          </div>,
          <div
            className="github-cards"
          >
            <GithubCard
              card={
                Object {
                  "cardDescription": "This is lots of long text from the body of my issue",
                  "cardNumber": 23,
                  "cardStatus": "Open",
                  "cardTitle": "This is the title",
                  "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
                }
              }
            />
            <GithubCard
              card={
                Object {
                  "cardDescription": "Description 2",
                  "cardNumber": 16,
                  "cardStatus": "Open",
                  "cardTitle": "This is title2",
                  "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/16",
                }
              }
            />
          </div>,
        ],
        "className": "github-cards-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="github-title"
              >
                Github Issues
              </div>,
              <button
                onClick={[Function]}
              >
                status
              </button>,
              null,
            ],
            "className": "github-cards-container-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Github Issues",
                "className": "github-title",
              },
              "ref": null,
              "rendered": "Github Issues",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "status",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "status",
              "type": "button",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <GithubCard
                card={
                  Object {
                    "cardDescription": "This is lots of long text from the body of my issue",
                    "cardNumber": 23,
                    "cardStatus": "Open",
                    "cardTitle": "This is the title",
                    "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
                  }
                }
              />,
              <GithubCard
                card={
                  Object {
                    "cardDescription": "Description 2",
                    "cardNumber": 16,
                    "cardStatus": "Open",
                    "cardTitle": "This is title2",
                    "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/16",
                  }
                }
              />,
            ],
            "className": "github-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "card": Object {
                  "cardDescription": "This is lots of long text from the body of my issue",
                  "cardNumber": 23,
                  "cardStatus": "Open",
                  "cardTitle": "This is the title",
                  "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "card": Object {
                  "cardDescription": "Description 2",
                  "cardNumber": 16,
                  "cardStatus": "Open",
                  "cardTitle": "This is title2",
                  "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/16",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
